scipy/optimize/_highs/setup.py

```diff
@@ -0,0 +1,133 @@
+
+import sys
+import pathlib
+from datetime import datetime
+
+
+def _get_sources(CMakeLists, start_token, end_token):
+    # Read in sources from CMakeLists.txt
+    CMakeLists = pathlib.Path(__file__).parent / CMakeLists
+    with open(CMakeLists, 'r') as f:
+        s = f.read()
+
+        # Find block where sources are listed
+        start_idx = s.find(start_token) + len(start_token)
+        end_idx = s[start_idx:].find(end_token) + len(s[:start_idx])
+        sources = s[start_idx:end_idx].split('\n')
+        sources = [s.strip() for s in sources if s[0] != '#']
+
+    # Make relative to setup.py
+    sources = [str(pathlib.Path('src/' + s)) for s in sources]
+    return sources
+
+
+# Grab some more info about HiGHS from root CMakeLists
+def _get_version(CMakeLists, start_token, end_token=')'):
+    CMakeLists = pathlib.Path(__file__).parent / CMakeLists
+    with open(CMakeLists, 'r') as f:
+        s = f.read()
+        start_idx = s.find(start_token) + len(start_token) + 1
+        end_idx = s[start_idx:].find(end_token) + len(s[:start_idx])
+    return s[start_idx:end_idx].strip()
+
+
+def configuration(parent_package='', top_path=None):
+
+    from numpy.distutils.misc_util import Configuration
+    config = Configuration('_highs', parent_package, top_path)
+
+    # HiGHS info
+    HIGHS_VERSION_MAJOR = _get_version('CMakeLists.txt', 'HIGHS_VERSION_MAJOR')
+    HIGHS_VERSION_MINOR = _get_version('CMakeLists.txt', 'HIGHS_VERSION_MINOR')
+    HIGHS_VERSION_PATCH = _get_version('CMakeLists.txt', 'HIGHS_VERSION_PATCH')
+    GITHASH = 'n/a'
+    HIGHS_DIR = str(pathlib.Path(__file__).parent.resolve())
+
+    # Here are the pound defines that HConfig.h would usually provide;
+    # We provide an empty HConfig.h file and do the defs and undefs
+    # here:
+    TODAY_DATE = datetime.today().strftime('%Y-%m-%d')
+    # ('OPENMP', None), ?
+    DEFINE_MACROS = [
+        ('CMAKE_BUILD_TYPE', '"Release"'),
+        ('HiGHSRELEASE', None),
+        ('IPX_ON', 'ON'),
+        ('HIGHS_GITHASH', '"%s"' % GITHASH),
+        ('HIGHS_COMPILATION_DATE', '"' + TODAY_DATE + '"'),
+        ('HIGHS_VERSION_MAJOR', HIGHS_VERSION_MAJOR),
+        ('HIGHS_VERSION_MINOR', HIGHS_VERSION_MINOR),
+        ('HIGHS_VERSION_PATCH', HIGHS_VERSION_PATCH),
+        ('HIGHS_DIR', '"' + HIGHS_DIR + '"'),
+    ]
+    UNDEF_MACROS = [
+        'OPENMP',  # unconditionally disable openmp
+        'EXT_PRESOLVE',
+        'SCIP_DEV',
+        'HiGHSDEV',
+        'OSI_FOUND',
+    ]
+
+    # HiGHS written using C++11 standard, we use C++14 for std::make_unique;
+    # -O3 is used for all libraries in HiGHS
```

Comment:
These flags normally come from `numpy.distutils`, and may vary by platform/compiler. Best to not specify them here unless there's a very good reason. And if so (e.g. the `/O2`'s below) please comment on the reason for hardcoding.

Reply:
I would love to use `numpy.distutils` to set these flags, I did not see how to [docs](https://docs.scipy.org/doc/numpy/reference/distutils.html#module-numpy.distutils), but I could have missed it.  The goal is to use optimization flags consistent with those in upstream HiGHS build.

Can you provide an example of how to set flags using `numpy.distutils`?  I've looked at a few `setup.py` files in scipy and haven't come across this.

Reply:
Yeah, those docs aren't very useful for most things unfortunately. Those flags are basically hardcoded per compiler, for example https://github.com/numpy/numpy/blob/master/numpy/distutils/intelccompiler.py#L21. There's no API to set them.

> The goal is to use optimization flags consistent with those in upstream HiGHS build.

Looks like it is valid to provide multiple `-O` flags (at least for GCC, not sure in general), and the last one takes precedence.

Not sure if this is the right goal or not to be honest, we don't do this for any other vendored code.

Reply:
I see the issue then.  Let's conform to scipy for now and remove them.  I'm happy with this solution as long as unit tests keep passing and we get the same answers as upstream HiGHS.

Reply:
`c++11` flags are still hardcoded - is this the correct approach?

Reply:
You'll want to use `get_cxx_std_flag` as is done in a couple of other places (example: `scipy/spatial/setup.py`).

Reply:
Fixed in commit [29ee35ff47325ce3fadd958d4be0af0fd9762bc7]( https://github.com/mdhaber/scipy/pull/5/commits/29ee35ff47325ce3fadd958d4be0af0fd9762bc7)